backend/app/schemas/knowledge.py

```
@@ -112,6 +112,32 @@ class KnowledgeBaseUpdate(BaseModel):
+        None,
+        ge=2,
+        le=50,
+        description="Maximum number of knowledge base tool calls allowed per conversation",
+    )
+    exempt_calls_before_check: Optional[int] = Field(
+        None,
+        ge=1,
+        description="Number of calls exempt from token checking (must be < max_calls_per_conversation)",
+    )
+
+    @model_validator(mode="after")
```

Ruff (0.14.14)

Undefined name `model_validator`

(F821)